plugins/modules/github_gpg_key.py

```diff
@@ -0,0 +1,386 @@
+            data=json.dumps({'name': name, 'raw_key': armored_public_key})).json()
+
+
+def delete_keys(session, to_delete, check_mode):
+    if check_mode:
+        return
+
+    for key in to_delete:
+        session.request('DELETE', API_BASE + '/user/keys/%s' % key["id"])
+
+
+def ensure_key_absent(session, name, check_mode):
+    to_delete = [key for key in get_all_keys(session) if key['name'] == name]
+    delete_keys(session, to_delete, check_mode=check_mode)
+
+    return {'changed': bool(to_delete),
+            'deleted_keys': to_delete}
+
+
+def ensure_key_present(module, session, name, armored_public_key, force, check_mode):
+    all_keys = get_all_keys(session)
+    matching_keys = [k for k in all_keys if k['name'] == name]
+    deleted_keys = []
+
+    new_signature = armored_public_key
+    for key in all_keys:
+        existing_signature = key['raw_key']
+        if new_signature == existing_signature and key['name'] != name:
+            module.fail_json(msg=(
+                "another key with the same content is already registered "
+                "under the name |{0}|").format(key['title']))
+
+    if matching_keys and force and matching_keys[0]['raw_key'] != new_signature:
+        delete_keys(session, matching_keys, check_mode=check_mode)
+        (deleted_keys, matching_keys) = (matching_keys, [])
+
+    if not matching_keys:
+        key = create_key(session, name, armored_public_key, check_mode=check_mode)
+    else:
+        key = matching_keys[0]
+
+    return {
+        'changed': bool(deleted_keys or not matching_keys),
+        'deleted_keys': deleted_keys,
+        'matching_keys': matching_keys,
+        'key': key
+    }
+
+
+def run_module(module, token, name, armored_public_key, state, force, check_mode):
+    session = GitHubSession(module, token)
+    if state == 'present':
+        ensure_key_present(module, session, name, armored_public_key, force=force,
+                           check_mode=check_mode)
+    elif state == 'absent':
+        result = ensure_key_absent(session, name, check_mode=check_mode)
+
+    return result
+
+
+def main():
+    argument_spec = {
+        'token': {'required': True, 'no_log': True},
+        'name': {'required': True},
+        'armored_public_key': {},
+        'state': {'choices': ['present', 'absent'], 'default': 'present'},
+        'force': {'default': True, 'type': 'bool'},
+    }
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    armored_public_key = module.params.get('armored_public_key')
+
+    if armored_public_key:
+        armored_public_key_parts = armored_public_key.split("\r\n")
+        armored_public_key_start = "-----BEGIN PGP PUBLIC KEY BLOCK-----"
+        armored_public_key_end = "-----END PGP PUBLIC KEY BLOCK-----"
+        if armored_public_key_parts[0] != armored_public_key_start or \
+        armored_public_key_parts[-1] != armored_public_key_end:
```

Comment:
Maybe move this to a function `def is_armored_public_key(candidate):` that's simply called here.